http-client-openssl/test/Spec.hs

```diff
@@ -13,6 +13,26 @@ main = withOpenSSL $ hspec $ do
         withResponse (parseRequest_ "HEAD https://s3.amazonaws.com/hackage.fpcomplete.com/01-index.tar.gz") manager $ \res -> do
             responseStatus res `shouldBe` status200
             lookup "content-type" (responseHeaders res) `shouldBe` Just "application/x-gzip"
+
```

Comment:
Any objection to removing these tests for `http-client-openssl`?

Reply:
@snoyberg What's the motivation for removing them?

Reply:
The tests fail because openssl isn't being configured to perform these checks. Meanwhile, http-client-tls _is_ performing these checks, the tests pass, and would prevent against regressions in the future.

Reply:
Thanks for explaining. Let's remove the tests then.

Reply:
OK, let's try with https://github.com/snoyberg/http-client/pull/423